File: task-manager-system/backend/Models/Task.cs
namespace Backend.Models;

public enum TaskStatus
{

    ToDo = 1,

    InProgress,

    Done

}

public enum TaskPriority
{

    NoPriority,

    LowPriority,

    MediumPriority,

    HighPriority,

    Critical

}

public class Task
{

    public int Id { get; init; }

    public int? AuthorId { get; init; }

    public User? Author { get; init; }

    public string? Title { get; set; }

    public string? Description { get; set; }

    public DateOnly Deadline { get; set; }

    public int? AssigneeId { get; set; }

    public User? Assignee { get; set; }

    public TaskStatus Status { get; set; }

    public TaskPriority Priority { get; set; }

}

File: task-manager-system/frontend/src/app/models/task.ts
import { TaskPriority } from "../enums/task-priority";
import { TaskStatus } from "../enums/task-status";
import { User } from "./user";

export class Task {

    public readonly id!: number;

    public readonly author!: User;

    public title!: string;

    public description?: string;

    public deadline!: string;

    public assignee?: User;

    public status!: TaskStatus;

    public priority!: TaskPriority;

}